src/openldap/mod.rs

```diff
@@ -231,6 +233,48 @@ impl OpenLDAP {
             .insert("LDAP_PASSWORD_HASH".to_owned(), password_hash.to_string());
         self
     }
+
+    /// Sets a custom ldif file (content) which should be used.
+    /// Default: `[]`
+    pub fn with_ldif_file(mut self, source: impl Into<CopyDataSource>) -> Self {
+        let n = self.copy_to_sources.len() + 1;
+
+        let container_ldif_path = format!("/ldifs/custom{n}.ldif");
+
+        self.copy_to_sources
+            .push(CopyToContainer::new(source, container_ldif_path));
+        self
+    }
+
+    /// Set all necessary certificate artifacts to build up a secure communication.
+    /// Default: `[]`
+    pub fn with_certs(
+        mut self,
+        cert: impl Into<CopyDataSource>,
+        key: impl Into<CopyDataSource>,
+        ca: impl Into<CopyDataSource>,
+    ) -> Self {
```

Comment:
probably it makes sense to split them into separate methods? (`with_cert`, `with_cert_key`, `with_cert_ca`)
Not sure about openlap requirements, but `ca` might be an optional one for example. 

However it also enables TLS, that makes sense.

WDYT of `with_tls(cert, key`) + `with_cert_ca(ca)` ? 🤔